src/button.tsx

```diff
@@ -217,6 +221,59 @@ const useDangerButtonStyles = createUseStyles((theme) => ({
     inverse: dangerButtonStyles(theme),
 }));
 
+const renderButtonContent = ({
+    content,
+    defaultIconSize,
+    renderText,
+}: {
+    content: React.ReactNode;
+    defaultIconSize: number;
+    renderText: (text: React.ReactNode) => React.ReactNode;
+}): React.ReactNode => {
+    const length = React.Children.count(content);
+    const childrenArr = React.Children.toArray(content);
+    const resultChildrenArr: Array<React.ReactNode> = [];
+    let accText: Array<React.ReactNode> = [];
+    const flushAccText = () => {
+        resultChildrenArr.push(
+            <React.Fragment key={resultChildrenArr.length}>{renderText(accText)}</React.Fragment>
+        );
+        accText = [];
+    };
+    childrenArr.forEach((element, idx) => {
+        const isFirstChild = idx === 0;
+        const isLastChild = idx === length - 1;
+
+        const isIconElement = React.isValidElement(element);
+
+        if (isIconElement) {
+            if (accText.length) {
+                flushAccText();
+            }
+            const sizeInPx = element.props.size ?? defaultIconSize;
+            resultChildrenArr.push(
+                <div
+                    key={resultChildrenArr.length}
+                    style={{
```

Comment:
just for understanding purposes: why the style in place of a class in this case?

Reply:
`marginLeft` and `marginRight` are dynamic styles, so it's easier and more convenient to use `style` here, although I could have 3 different classNames and combine them too